src/components/inputModal.tsx

```diff
@@ -1,27 +1,162 @@
+              <Form.Control
+                type="text"
+                placeholder="Enter name"
+                {...register("name")}
+              />
+              {errors.name && (
+                <Form.Text className="text-danger">
+                  {errors.name.message?.toString()}
+                </Form.Text>
+              )}
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="marks">
+              <Form.Label>Marks</Form.Label>
+              <Form.Control
+                type="number"
+                placeholder="Enter marks obtained"
+                {...register("marks")}
+              />
+              {errors.marks && (
+                <Form.Text className="text-danger">
+                  {errors?.marks?.message?.includes(
+                    "marks must be a `number` type"
+                  )
+                    ? "Must enter marks"
+                    : errors?.marks?.message?.toString()}{" "}
+                </Form.Text>
+              )}
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="subject">
+              <Form.Label>Subject</Form.Label>
+              <Form.Select aria-label="Select Subject" {...register("subject")}>
+                <option value="English">English</option>
+                <option value="Math">Math</option>
+                <option value="Science">Science</option>
+              </Form.Select>
+              {errors.subject && (
+                <Form.Text className="text-danger">
+                  Must choose a Subject
+                </Form.Text>
+              )}
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="grades">
+              <Form.Label>Grades</Form.Label>
+              <Form.Select aria-label="Select grade" {...register("grades")}>
+                <option value="A">A</option>
+                <option value="B">B</option>
+                <option value="C">C</option>
+                <option value="D">D</option>
+                <option value="F">F</option>
+              </Form.Select>
+              {errors.grades && (
+                <Form.Text className="text-danger">
+                  Must choose a Grade
+                </Form.Text>
+              )}
+            </Form.Group>
+            <div className="inputForm-btnGroup">
+              <Button
+                style={{
+                  paddingLeft: "60px",
+                  paddingRight: "60px",
+                  backgroundColor: "white",
+                  color: "black",
+                }}
+                onClick={() => props.setVisible(false)}
+              >
+                Close
+              </Button>
+              <Button
+                style={{
+                  paddingLeft: "60px",
+                  paddingRight: "60px",
+                  backgroundColor: "#4aaa9a",
+                }}
+                type="submit"
+              >
+                Add
+              </Button>
```

Comment:
you can create a generic button component that handles both primary and secondary buttons. you can create common components in components.common folder